test-suite/js/src/test/scala/org/scalajs/testsuite/typedarray/TypedArrayTest.scala

```diff
@@ -36,6 +40,32 @@ trait TypedArrayTest[V, T <: TypedArray[V, T]] {
   def hasType(obj: Any): Boolean
   def intToV(n: Int): V
 
+  @Test def should_provide_a_factory_method_of(): Unit = {
+    val x = ofFn(intToV(0), intToV(1), intToV(2))
+    assertEquals(3, x.length)
+    assertEquals(2, x(2))
+  }
+
+  @Test def should_provide_a_factory_method_from(): Unit = {
+    val x = fromFn(js.Array(intToV(0), intToV(1), intToV(2)))
+    assertEquals(3, x.length)
+    assertEquals(2, x(2))
+  }
+
+  @Test def should_provide_a_factory_method_from_with_mapping_function(): Unit = {
+    val src = js.Array("", "a", "bc")
+    val x = fromFn(src)(s => intToV(s.length))
```

Comment:
You'll need an explicit type for `s` here for this to compile under Scala 2.10 and 2.11.